lib/engine/src/export.rs

```diff
@@ -49,12 +52,14 @@ impl From<VMExport> for Export {
 #[derive(Debug, Clone, PartialEq)]
 pub struct ExportFunction {
     /// The VM function, containing most of the data.
-    pub vm_function: VMExportFunction,
+    pub vm_function: Arc<VMExportFunction>,
     /// Function pointer to `WasmerEnv::init_with_instance(&mut self, instance: &Instance)`.
     ///
     /// This function is called to finish setting up the environment after
     /// we create the `api::Instance`.
     pub import_init_function_ptr: Option<ImportInitializerFuncPtr>,
+    /// The destructor to free the host environment.
+    pub host_env_drop_fn: Option<fn(*mut std::ffi::c_void)>,
```

Comment:
A function can be exported twice. How do you verify that `host_env_drop_fn` is only called once?

Reply:
Do we have any tests for this? I believe this should behave the same as initialization, which should also only run once